src/dstore/pmix_esh.c

```diff
@@ -701,29 +1024,42 @@ static int _pmix_getpagesize(void)
 #endif
 }
 
-static seg_desc_t *_create_new_segment(segment_type type, char *nsname, uint32_t id)
+static seg_desc_t *_create_new_segment(segment_type type, const char *nsname, uint32_t id)
 {
     int rc;
     char file_name[PMIX_PATH_MAX];
     size_t size;
     seg_desc_t *new_seg = NULL;
+    char *nspace_path = NULL;
+    uid_t jobuid;
 
     PMIX_OUTPUT_VERBOSE((10, pmix_globals.debug_output,
                          "%s:%d:%s: segment type %d, nspace %s, id %u",
                          __FILE__, __LINE__, __func__, type, nsname, id));
 
+    nspace_path =  _esh_nspace_path(nsname);
```

Comment:
this is an example of multiple searches: you have to search the namespace twice: for `_esh_nspace_path` and `_esh_nspace_jobuid`,